dora/core/server/master/src/main/java/alluxio/master/job/DoraLoadJob.java

```diff
@@ -225,28 +185,34 @@ public URIStatus next() {
    */
   public List<DoraLoadTask> prepareNextTasks() {
     LOG.debug("Preparing next set of tasks for jobId:{}", mJobId);
-    ImmutableList.Builder<URIStatus> batchBuilder = ImmutableList.builder();
+    ImmutableList.Builder<UfsStatus> batchBuilder = ImmutableList.builder();
     int i = 0;
     int startRetryListSize = mRetryFiles.size();
     int filesToLoad = 0;
     while (filesToLoad < RETRY_THRESHOLD
-        && i++ < startRetryListSize
-        && mRetryFiles.peek() != null) {
+        && i++ < startRetryListSize && mRetryFiles.peek() != null) {
       String path = mRetryFiles.poll();
       try {
-        URIStatus uriStatus = mFs.getStatus(new AlluxioURI(path));
+        UfsStatus uriStatus = mUfs.getStatus(path);
         batchBuilder.add(uriStatus);
         ++filesToLoad;
-      } catch (AlluxioException | IOException ex) {
-        if (!(ex instanceof FileDoesNotExistException)) {
+      } catch (IOException e) {
+        // The previous list or get might contain stale file metadata.
+        // For example, if a file gets removed before the worker actually loads it,
+        // the load will fail and the scheduler will retry.
+        // In such case, a FileNotFoundException might be thrown when we attempt to
+        // get the file status again, and we simply ignore that file.
+        // TODO(elega) this might result in an unlimited retry.
```

Comment:
this to do can be done by adding up the failure conter

Reply:
yeah you are correct. I removed this comment. If the failure count exceeds the threshold, the job will just fail and the scheduler will stop running new tasks.